src/core/StelCore.cpp

```diff
@@ -1351,40 +1351,55 @@ void StelCore::moveObserverTo(const StelLocation& target, double duration, doubl
 	emit locationChanged(getCurrentLocation());
 }
 
+QCache<size_t, qint64> StelCore::utcOffsetCache;
+
 double StelCore::getUTCOffset(const double JD) const
 {
-	int year, month, day, hour, minute, second;
-	StelUtils::getDateTimeFromJulianDay(JD, &year, &month, &day, &hour, &minute, &second);
+	// This method takes a significant amount of time. Try to cache a few entries.
+	// All these elements can influence UTCOffset, so we must include them in the cache.
+	// Presumably, offset does not change during a minute, so we can exclude seconds here.
```

Comment:
Looks like an unsafe assumption to me. What is the cost of keeping the seconds?

Reply:
updates every second, not every minute.